File: exercice4/bank.java
class Playground {
    public static void main(String[ ] args) {
        Client stephane = new Client ("Stéphane", "4854FZFAZ546FZ");
        ProClient marie = new ProClient ("Marie", "186DD5978FDZ", "78948923600032");
        stephane.setName("Nicolas");
        System.out.println(stephane.getName());
        System.out.println("Le client " + marie.name + " possède le compte numéro " + marie.accountNB + " et son numéro de SIRET est le : " + marie.siret);
    }
}

public class Client {
    protected String name;
    protected String accountNB;

    public Client(String name, String accountNB){
        this.name = name;
        this.accountNB = accountNB;
    }

    String getAccountNB(){
        return this.accountNB;
    }

    void setName(String name){
      this.name = name;
      return;
    }

    String getName(){
        return this.name;
    }
}

class ProClient extends Client {

  String siret;

  ProClient (String name, String accountNB,String siret) {
    super(name, accountNB);
    this.siret = siret;
  }
}

File: exercice4/bank.ts
class Client {
  public name: string;
  public accountNb: string;

  public constructor(name: string, accountNb: string) {
    this.name = name;
    this.accountNb = accountNb;
  }

  public getAccountNb(): string {
    return this.accountNb;
  }

  public getName(): string {
    return this.name;
  }

  public setName(name: string): void {
    this.name = name;
  }
}

class ProClient extends Client {

  siret: string;

  constructor(name: string, accountNb: string, siret: string) {
    super(name, accountNb);
    this.siret = siret;
  }

}

let marie: ProClient = new ProClient("Marie", "186DD5978FDZ", "789 489 236 00032")

let stephane: Client = new Client("Stéphane", "14DFG78FAF78");
console.log(stephane.getAccountNb());

stephane.setName("Nicolas");
console.log(stephane.getName());

console.log(marie.getName())
